consensus/reactor_old_propagation_test.go

```diff
@@ -0,0 +1,141 @@
+package consensus
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	cfg "github.com/cometbft/cometbft/config"
+)
+
+func TestReactorDisableOldPropagationRoutineOnAppVersion5(t *testing.T) {
```

Comment:
this test is insanely complicated

after refactoring for the other changes, be sure to write a simpler lower level test.

after the refactor to move the check to after the commit, look at the tests for committing a block. do what they do to simply change the app version and then check if it was changed globally. That's all that's needed. The test must be maintainable this is not